### PR TITLE
Don't refresh entire eviction pool on every eviction

### PR DESCRIPTION
**Pull request by tylerwilliams**
Profiling + metrics show that eviction can be a bottleneck. A major contributor is refreshing the atime of every item in the pool on every eviction.

This CL stops doing that -- instead the approxlru will refresh an item before deleting and skip deletion if the sample's atime differs between the pool value and the live value.

### CODE REVIEW

File: server/util/approxlru/approxlru.go [hunk 1]
```diff
@@ -250,13 +235,30 @@ func (l *LRU[T]) resampleK(k int) error {
...
 		l.mu.Lock()
 		oldLocalSizeBytes := l.localSizeBytes
 		oldGlobalSizeBytes := l.globalSizeBytes
 		l.mu.Unlock()
 
-		log.Infof("Evictor attempting to evict %q (last accessed %s)", sample.Key, time.Since(sample.Timestamp))
-		skip, err := l.onEvict(l.ctx, sample)
+		skip, timestamp, err := l.onRefresh(l.ctx, sample.Key)
+		if err != nil {
+			log.Warningf("Could not refresh timestamp for %q: %s", sample.Key, err)
```
Comment (vadimberezniker) [inline server/util/approxlru/approxlru.go:246]: Infof

File: server/util/approxlru/approxlru.go [hunk 2]
```diff
@@ -250,13 +235,30 @@ func (l *LRU[T]) resampleK(k int) error {
...
+			continue
+		}
+		if skip {
+			continue
+		}
+		if sample.Timestamp != timestamp {
+			log.Warningf("Evictor skipping %q; atime has changed %s -> %s", sample.Key, sample.Timestamp, timestamp)
```
Comment (vadimberezniker) [inline server/util/approxlru/approxlru.go:253]: Infof

**Review by vadimberezniker — APPROVED** [review]

File: server/util/approxlru/approxlru.go (hunk 1 above)
Comment (tylerwilliams) [reply server/util/approxlru/approxlru.go]: Done.

File: server/util/approxlru/approxlru.go (hunk 2 above)
Comment (tylerwilliams) [reply server/util/approxlru/approxlru.go]: Done.

**Pull request merged**